onnxruntime/core/session/inference_session.cc

```diff
@@ -1151,9 +1151,17 @@ common::Status InferenceSession::Initialize() {
     // need to keep the initializers if we're going to save the optimized model
     bool keep_initializers = !session_options_.optimized_model_filepath.empty();
 
+    // we only want to use the serialized session state in a minimal build.
+    // in a full build, even if we loaded from an ORT format model, optimizers may have run and changed the
+    // graph, which would result in it no longer matching the saved session state.
+    // as it's a full build we can create the SessionState as per usual using kernel lookups, so we can take the safe
+    // option and ignore the serialized session state.
+    const experimental::fbs::SessionState* serialized_session_state = nullptr;
+#if defined(ORT_MINIMAL_BUILD)
     auto* serialized_session_state = !ort_format_model_bytes_.empty()
```

Comment:
> auto* [](start = 3, length = 6)

Remove?

Reply:
I see the problem(multiple definitions) when compiling with ORT_MINIMAL_BUILD.

---
In reply to: [521617571](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521617571) [](ancestors = 521617571)